sapl/sessao/views.py

```diff
@@ -781,6 +782,27 @@ def get_initial(self):
 
                 messages.add_message(self.request, messages.ERROR, msg)
                 return {}
+            
+    class DetailView(Crud.DetailView):
+        def get_id(self, url):
+            u_pars = urlparse(url)
+            quer_v = parse_qs(u_pars.query).get('v')
+            if quer_v:
+                return quer_v[0]
+            path = u_pars.path.split('/')
+            if path:
+                return path[-1]
+
+        def get_context_data(self, **kwargs):
+            context = super().get_context_data(**kwargs)
+            if context['object'].url_video:
```

Comment:
Primeiramente, atribui `context['object'].url_video` a uma variável para melhorar a legibilidade a partir da linha 798.

A gente tem um pequeno detalhe aqui: existem várias opções de vídeo:
* um arquivo mp4/ogg/flv que referenciado por uma URL;
* um vídeo no Facebook
* um vídeo no Youtube. 

Vamos ter que analisar a URL de vídeo e daí adequar de acordo porque se for um vídeo do FB ou YT teria que encapsular num iframe, mas se for um arquivo mp4 referenciado por uma URL então teria que usar a tag <video> do HTML5. 

Ademais, acho melhor colocar essa lógica no template `crud/detail.html` do que enxertá-lo aqui.

Reply:
Exemplos de URLs de teste:

* https://www.youtube.com/watch?v=W_TlhUAfqDw
* https://www.facebook.com/plugins/video.php?href=https%3A%2F%2Fwww.facebook.com%2FCamaraMedicilandiaPA%2Fvideos%2F2228456237418480%2F&show
* http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4

Reply:
Uma solução quick-and-dirty:
No arquivo `common_tags.py` adicione e modifique os arquivos da forma abaixo (adicionando suas checagens se existe o parâmetro `v`, por exemplo)
```
@register.filter
def video_url(value):
    return url(value) and value.endswith("mp4")

@register.filter
def youtube_url(value):
    import re
    # Test if YouTube video
    # tested on https://pythex.org/
    youtube_pattern = "^(http://?|https://?www\.?youtube\.com/watch\?v=)"
    r = re.findall(youtube_pattern, value)
    return True if r else False

@register.filter
def get_youtube_id(value):
    from urllib.parse import urlparse, parse_qs
    u_pars = urlparse(value)
    quer_v = parse_qs(u_pars.query).get('v')[0]
    return quer_v
```

E no artigo `templates/crud/detail.html` a partir da linha 75 as linhas abaixo:
```
                      {% elif column.text|video_url %}
                        <div class="form-control-static">
                            <video width="320" height="120" controls>
                                <source src="{{ column.text|safe }}" type="video/{{ column.text|file_extension }}">
                                <p>Este navegador não suporta o elemento vídeo.</p>
                            </video>
                        </div>
                      {% elif column.text|youtube_url %}
                          <iframe id="ytplayer" type="text/html" width="420" height="230"
                          src="http://www.youtube.com/embed/{{ column.text|get_youtube_id }}"
                          frameborder="0"/>
                      {% elif column.text|url %}
                        <div class="form-control-static"><a href="{{ column.text|safe }}"> {{ column.text|safe|default:"" }} </a></div>
                      {% else %}
                        <div class="form-control-static">{{ column.text|safe|default:"" }}</div>
                      {% endif %}
```

Reply:
Foi seguido essa ideia, acresentando videos do facebook porem usando o jeito recomendado pela pagina do facebook, que não enquadra o link posto nos exemplos.

Exemplo de link aceito:''' https://www.facebook.com/facebook/videos/new-ways-to-watch-facebook-video/10155517557066729/